File: web/src/types/snippet.ts
export interface Snippet {
  id: string
  name: string
  content: string
  description: string
  tags: string[]
  is_global: boolean
  created_at: string
  updated_at: string
}

export interface SnippetCreateRequest {
  name: string
  content: string
  description?: string
  tags?: string[]
  is_global?: boolean
}

export interface SnippetUpdateRequest {
  name?: string
  content?: string
  description?: string
  tags?: string[]
  is_global?: boolean
}

File: server/model/snippet.go
package model

import "time"

type Snippet struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Content     string    `json:"content"`
	Description string    `json:"description,omitempty"`
	Tags        []string  `json:"tags"`
	IsGlobal    bool      `json:"is_global"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type SnippetCreateRequest struct {
	Name        string   `json:"name"`
	Content     string   `json:"content"`
	Description string   `json:"description,omitempty"`
	Tags        []string `json:"tags,omitempty"`
	IsGlobal    *bool    `json:"is_global,omitempty"`
}

type SnippetUpdateRequest struct {
	Name        *string  `json:"name,omitempty"`
	Content     *string  `json:"content,omitempty"`
	Description *string  `json:"description,omitempty"`
	Tags        []string `json:"tags,omitempty"`
	IsGlobal    *bool    `json:"is_global,omitempty"`
}
